src/common/file-utils.cpp

```diff
@@ -109,6 +109,19 @@ rnp_fopen(const char *filename, const char *mode)
 #endif
 }
 
+int
+rnp_access(const char *path, int mode)
+{
+#ifdef _WIN32
+    try {
+        return _waccess(wstr_from_utf8(path).c_str(), mode);
```

Comment:
Should we mask 0x1 bit as only values 0, 2, 4 and 6 are supported on Windows according to 
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/access-waccess?view=msvc-160
Also, the docs say, "it does not check the filesystem security settings",
meaning that if the directory needs administrative permissions `rnp_access` may return success.
Should we consider this situation?

Reply:
I think removing X_OK definition on Windows should be enough? So only supported values are defined as macros.

About second part,
looks like WinAPI implemenation of this task is bit complicated,
I've found some guide here
https://blog.aaronballman.com/2011/08/how-to-check-access-rights/
but in comments there are reports that it does not properly work on Windows 8 and newer...

Reply:
And some Windows-specific test case must be added then,
where test home directory exists, but write access is not allowed for
Windows user running tests. This will require admin privileges...

Reply:
@rrrooommmaaa @ni4 @ronaldtse should we implement checking Windows security permissions or keep as is?

Reply:
@antonsviridenko Actually it's not just for tests - we should make sure that if user runs rnp (via FFI or CLI) on non-writeable home directory, it handles this case gracefully. I think this could be extracted to the separate task.

Reply:
Actually, the only test of missing permissions for Unix directory gives quite a headache in CI tests (especially in docker containers). We need to create a special user there. Maybe it's enough to test it locally on WIndows?

Reply:
@rrrooommmaaa We should make sure that this case is handled correctly, i.e. not crash/whatever else in this case. Adding user is just a few lines in GHA file, don't think this is a problem.

Reply:
Is it ok to use WinAPI functions here?

Reply:
ping @ni4 ^^
looks like we'll need to add linking to advapi32.dll
in order to use AccessCheck() and other related WinAPI functions

Reply:
@antonsviridenko Sorry, missed your previous comment. I think it would be enough to keep it as it is now, without adding new dependencies and complications. Probably I used incorrect wording in previous comments - I meant, that we should make sure that RNP works ok in case if _waccess() returns 'writeable', but that requires administrative permissions or so on. Basically, even if we'd implement all complicated checks, we cannot 100% guarantee that write to the directory will be allowed.